test/jdk/javax/swing/JButton/4659800/bug4659800.java

```diff
@@ -0,0 +1,147 @@
+ * ActionEvent on focused Button or not.
+ * @run main bug4659800
+ */
+public class bug4659800 {
+    private static JFrame frame = null;
+    private static volatile boolean buttonPressed = false;
+    private boolean testFailed = false;
+    private JButton button1;
+    private Robot robot;
+    private JButton dummyButton;
+
+    public static void main(String[] s) {
+        bug4659800 test = new bug4659800();
+        try {
+            test.runTest();
+        } finally {
+            if (frame != null) {
+                frame.dispose();
+                frame = null;
+            }
+        }
+
+    }
+
+    private static void setLookAndFeel(UIManager.LookAndFeelInfo laf) {
+        try {
+            UIManager.setLookAndFeel(laf.getClassName());
+        } catch (UnsupportedLookAndFeelException ignored) {
+            System.out.println("Unsupported L&F: " + laf.getClassName());
+        } catch (ClassNotFoundException | InstantiationException
+                | IllegalAccessException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void createUI() {
+        frame = new JFrame();
+        JPanel panel = new JPanel();
+        panel.setLayout(new FlowLayout());
+        button1 = new JButton("butt1");
+        buttonPressed = false;
+        button1.addActionListener(new ActionListener() {
+            public void actionPerformed(ActionEvent e) {
+                synchronized (this) {
+                    System.out.println("ActionPerformed on Button1 : ");
+                    buttonPressed = true;
+                    notifyAll();
+                }
+            }
+        });
+
+        panel.add(button1);
+
+        dummyButton = new JButton("butt2");
+        panel.add(dummyButton);
+        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        frame.setLayout(new FlowLayout());
+
+        frame.add(panel);
+
+        frame.setSize(200, 200);
+        frame.setLocationRelativeTo(null);
+        frame.setVisible(true);
+
+    }
+
+    public void runTest() {
+        try {
+            robot = new Robot();
+        } catch (Exception e) {
+            System.err.print("Error creating robot");
+            e.printStackTrace();
+            System.exit(1);
+        }
+        for (UIManager.LookAndFeelInfo laf : UIManager.getInstalledLookAndFeels()) {
+            System.out.println("Testing L&F: " + laf);
+
+            try {
+                SwingUtilities.invokeAndWait(() -> {
+                    setLookAndFeel(laf);
+                    createUI();
+                });
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+            // Clicks on dummy button to activate the parent window.
+            dummyButton.doClick();
+            button1.requestFocusInWindow();
+            Point p = button1.getLocationOnScreen();
+            robot.mouseMove(p.x + button1.getWidth() / 2, p.y + button1.getHeight() / 2);
```

Comment:
I don't understand why you need to programmatically press the `dummyButton` and then move the mouse cursor to `button1`.